### PR TITLE
Refactor LoadGen to use Library Mode, and add SRE Recipe LoadGen API placeholder

### PR DESCRIPTION
**Pull request by Gan-Tu**
In this PR, I

- added code to run LoadGen as a flask app, using Locust library mode code, listening on port 8089
- added a second Locust server (aka "environment") that will be started at the same time to serve as a decoupled, SRE Recipe API endpoint listening on port 8090
- added basic integration test for the SRE Recipe API endpoint in the e2e tests

I kept it so that functionally, the user facing web UI still supports the same set of existing tasks and behaviors, while the additional API endpoint will support a simple `/api/ping` endpoint as placeholder. 

I updated the Kubernetes and Docker file to expose the additional API port (8090) as port 81 (an arbitrary choice by me) on the main locust IP.

I will in future PRs complete the implementation of the SRE Recipe API endpoint. 

Thanks

---

Test it by deploying this change in GCP and try to access the `http://<locust ip>:81/api/ping`.

[Open in Cloud Shell](https://console.cloud.google.com/cloudshell/editor?cloudshell_git_repo=https://github.com/GoogleCloudPlatform/cloud-ops-sandbox.git&cloudshell_git_branch=gan-loadgen&shellonly=true&cloudshell_image=gcr.io/stackdriver-sandbox-230822/cloudshell-image/uncertified:v0.6.0&cloudshell_tutorial=docs/tutorial.md)

### CODE REVIEW

File: src/loadgenerator/Dockerfile
Comment (simonz130) [inline src/loadgenerator/Dockerfile:31]: curious, what is this port used for (8090)? Perhaps, worth adding a comment.

File: src/loadgenerator/logger.py
Comment (simonz130) [inline src/loadgenerator/logger.py:24]: @daniel-sanche can we use python-logging library for structured logging into stdout instead of having this json implementation?

File: loadgenerator-manifests/loadgenerator.yaml
Comment (daniel-sanche) [inline loadgenerator-manifests/loadgenerator.yaml:67]: Are we expecting users to find and use this UI? If so, we might want to expose it in a different way. (Maybe[ fan out ingress?](https://cloud.google.com/kubernetes-engine/docs/concepts/ingress#multiple_backend_services))

File: loadgenerator-manifests/loadgenerator.yaml
Comment (daniel-sanche) [inline loadgenerator-manifests/loadgenerator.yaml:35]: When are you planning on updating this? Before merging?

File: src/loadgenerator/logger.py
Comment (daniel-sanche) [reply src/loadgenerator/logger.py:24]: We could, although it wouldn't capture the timestamp in the exact same way.

What's the purpose of the logging here? Are users expected to see them in CLoud Logging, or is this just for internal use? Why does it need to be JSON formatted?

Comment (Gan-Tu) [reply src/loadgenerator/logger.py:24]: > What's the purpose of the logging here? Are users expected to see them in CLoud Logging, or is this just for internal use? Why does it need to be JSON formatted?

I copied this file exactly as it is from our other Python-based services (e.g. email, recommendation). I thought this is what is preferred in this project in terms of logging in python. I don't have a preference in terms of logging and whether they should be visible, but I think it is ok to be visible in Cloud logging. However, I will defer to you all to decide what loggings we want.

File: src/loadgenerator/Dockerfile
Comment (Gan-Tu) [reply src/loadgenerator/Dockerfile:31]: Sounds good. I added more documentation for the original ports exposed here as well

File: loadgenerator-manifests/loadgenerator.yaml
Comment (Gan-Tu) [reply loadgenerator-manifests/loadgenerator.yaml:67]: Users are not supposed to find this UI. It is only intended for access/use by SRE Recipe in terms of sending requests for generating specific traffic loads.

File: loadgenerator-manifests/loadgenerator.yaml
Comment (Gan-Tu) [reply loadgenerator-manifests/loadgenerator.yaml:35]: I don't have access to create a new image there. 

After this PR is approved, either someone can help me create one and I update it before merging, or we can have a different PR afterwards.

File: src/loadgenerator/logger.py
Comment (daniel-sanche) [reply src/loadgenerator/logger.py:24]: Ok, I think we should leave it as-is for this PR then. We could maybe open an issue to re-assess the logging method later, but I'm not sure I know the full scope

File: loadgenerator-manifests/loadgenerator.yaml
Comment (daniel-sanche) [reply loadgenerator-manifests/loadgenerator.yaml:35]: I think you can leave it as `gcr.io/stackdriver-sandbox-230822/sandbox/loadgenerator/gke:latest`, and the image will be [updated automatically when you push to master](https://github.com/GoogleCloudPlatform/cloud-ops-sandbox/blob/e1c33e6f6d7b8d43499942da52c762f94a3b20e7/.github/workflows/push-master.yml#L39), right?

**Review by daniel-sanche — APPROVED** [review]
LGTM, but you should revert the loadgenerator container path (and let the tests pass) before merging

File: loadgenerator-manifests/loadgenerator.yaml
Comment (Gan-Tu) [reply loadgenerator-manifests/loadgenerator.yaml:35]: sounds good!

Comment (Gan-Tu) [reply loadgenerator-manifests/loadgenerator.yaml:35]: @daniel-sanche 
If I change it to gcr.io/stackdriver-sandbox-230822/sandbox/loadgenerator/gke:latest, the E2E test for CI will run using the old latest image (because this PR hasn't been merged to master), but with the updated E2E test for the new API endpoint, thus fail. Should we send a follow up one line PR instead to change the path, once the new image is auto updated

Comment (daniel-sanche) [reply loadgenerator-manifests/loadgenerator.yaml:35]: The intention of the CI is that it should be automatically building and testing the image based on what's in the repo HEAD, not the latest published image. (see [here](https://github.com/GoogleCloudPlatform/cloud-ops-sandbox/blob/e1c33e6f6d7b8d43499942da52c762f94a3b20e7/.github/workflows/e2e-latest.yml#L54) where it builds a new tag, and [here](https://github.com/GoogleCloudPlatform/cloud-ops-sandbox/blob/e1c33e6f6d7b8d43499942da52c762f94a3b20e7/.github/workflows/e2e-latest.yml#L63) where it re-writes the manifests to use it)

Although looking at the code, it's [updating `./kubernetes-manifests`](https://github.com/GoogleCloudPlatform/cloud-ops-sandbox/blob/e1c33e6f6d7b8d43499942da52c762f94a3b20e7/.github/workflows/e2e-latest.yml#L63), but may be missing `loadgenerator-manifests`. You may have to update that part of the CI script if it's failing on your changes. But I don't think we should merge code to master pointing at a different container path

**Pull request merged**
